dss/api/bundles/__init__.py

```diff
@@ -187,8 +189,38 @@ def enumerate(replica: str,
     return response
 
 
+def project_uuid_from_file(replica, file):
+    with tracing.Subsegment('parameterization'):
+        handle = Config.get_blobstore_handle(replica)
+        bucket = replica.bucket
+
+    file_content = json.loads(
+        handle.get(
+            bucket,
+            "blobs/" + ".".join((
+                file[FileMetadata.SHA256],
+                file[FileMetadata.SHA1],
+                file[FileMetadata.S3_ETAG],
+                file[FileMetadata.CRC32C])
+            )
+        )
+    )
+    return file_content.get('provenance', {}).get('document_id', None)
+
+
+def determine_project_from_metadata(replica, files):
+    """
+    Return a project uuid from the metadata files or None if unsuccessful.
+
+    Relies on schema: https://schema.humancellatlas.org/type/protocol/5.1.0/protocol
+    """
+    for file in files:
+        if file['name'] in ('project.json', 'project_0.json'):
+            return project_uuid_from_file(replica, file)
+
+
 @dss_handler
-def put(uuid: str, replica: str, json_request_body: dict, version: str):
+def put(uuid: str, replica: str, json_request_body: dict, version: str, project: str = None):
```

Comment:
Let's just require `project` and believe whatever is passed in without validation.